File: apps/desktop/src-tauri/src/commands/window.rs
pub fn minimize(window: &tauri::WebviewWindow) -> Result<serde_json::Value, String> {
    window.minimize().map_err(|e| e.to_string())?;
    Ok(serde_json::Value::Null)
}

pub fn maximize(window: &tauri::WebviewWindow) -> Result<serde_json::Value, String> {
    if window.is_maximized().unwrap_or(false) {
        window.unmaximize().map_err(|e| e.to_string())?;
    } else {
        window.maximize().map_err(|e| e.to_string())?;
    }
    Ok(serde_json::Value::Null)
}

pub fn close(window: &tauri::WebviewWindow) -> Result<serde_json::Value, String> {
    window.close().map_err(|e| e.to_string())?;
    Ok(serde_json::Value::Null)
}

pub fn is_maximized(window: &tauri::WebviewWindow) -> Result<serde_json::Value, String> {
    let maximized = window.is_maximized().unwrap_or(false);
    Ok(serde_json::Value::Bool(maximized))
}

pub fn get_platform() -> Result<serde_json::Value, String> {
    let platform = match std::env::consts::OS {
        "macos" => "darwin",
        "windows" => "win32",
        other => other,
    };
    Ok(serde_json::Value::String(platform.to_string()))
}

pub fn get_home_dir() -> Result<serde_json::Value, String> {
    let home = dirs::home_dir()
        .map(|p| p.to_string_lossy().to_string())
        .unwrap_or_default();
    Ok(serde_json::Value::String(home))
}

/// Stub: requires Tauri dialog plugin (async). Will be implemented later.
pub fn select_directory() -> Result<serde_json::Value, String> {
    Ok(serde_json::Value::Null)
}

/// Stub: requires Tauri dialog plugin (async). Will be implemented later.
pub fn select_image_file() -> Result<serde_json::Value, String> {
    Ok(serde_json::Value::Null)
}

File: src/lib/trpc/routers/window.ts
import fs from "node:fs/promises";
import { homedir } from "node:os";
import path from "node:path";
import { getCurrentWindow } from "@tauri-apps/api/window";
import { open as openDialog } from "@tauri-apps/plugin-dialog";
import { z } from "zod";
import { publicProcedure, router } from "..";

export const createWindowRouter = () => {
	return router({
		minimize: publicProcedure.mutation(async () => {
			const window = getCurrentWindow();
			await window.minimize();
			return { success: true };
		}),

		maximize: publicProcedure.mutation(async () => {
			const window = getCurrentWindow();
			if (await window.isMaximized()) {
				await window.unmaximize();
			} else {
				await window.maximize();
			}
			return { success: true, isMaximized: await window.isMaximized() };
		}),

		close: publicProcedure.mutation(async () => {
			const window = getCurrentWindow();
			await window.close();
			return { success: true };
		}),

		isMaximized: publicProcedure.query(async () => {
			const window = getCurrentWindow();
			return await window.isMaximized();
		}),

		getPlatform: publicProcedure.query(() => {
			return process.platform;
		}),

		getHomeDir: publicProcedure.query(() => {
			return homedir();
		}),

		selectDirectory: publicProcedure
			.input(
				z
					.object({
						title: z.string().optional(),
						defaultPath: z.string().optional(),
					})
					.optional(),
			)
			.mutation(async ({ input }) => {
				const selected = await openDialog({
					directory: true,
					title: input?.title ?? "Select Directory",
					defaultPath: input?.defaultPath ?? undefined,
				});

				if (!selected) {
					return { canceled: true, path: null };
				}

				return { canceled: false, path: selected };
			}),

		selectImageFile: publicProcedure.mutation(async () => {
			const selected = await openDialog({
				title: "Select Organization Logo",
				filters: [
					{
						name: "Images",
						extensions: ["png", "jpg", "jpeg", "webp"],
					},
				],
			});

			if (!selected) {
				return { canceled: true, dataUrl: null };
			}

			const filePath = selected;
			const buffer = await fs.readFile(filePath);
			const ext = path.extname(filePath).slice(1).toLowerCase();
			const mimeType = ext === "jpg" ? "jpeg" : ext;
			const base64 = buffer.toString("base64");
			const dataUrl = `data:image/${mimeType};base64,${base64}`;

			return { canceled: false, dataUrl };
		}),
	});
};

export type WindowRouter = ReturnType<typeof createWindowRouter>;
